working/augmentation-libraries/feature-specification.md

```diff
@@ -318,6 +318,18 @@ It is a compile-time error if:
     augmentation can both declare a new declaration and augment it in the same
     file.*
 
+### Type inference
+
+Type inference for augmenting declarations works similar to overrides. Whenever
+a type is omitted, if the original declaration has a type specified, that type
+is inferred.
+
+This applies to return types, parameter types, and variable types, but not type
+parameters.
```

Comment:
That is, types, not names. 
What about type parameter bounds? Would be nice to inherit those.